activetable/activetable.py

```diff
@@ -0,0 +1,297 @@
+        'specify an explicit tolerance for.',
+        scope=Scope.content,
+        default=1.0,
+    )
+    max_score = Float(
+        display_name='Maximum score',
+        help='The number of points students will be awarded when solving all fields correctly.  '
+        'For partially correct attempts, the score will be pro-rated.',
+        scope=Scope.settings,
+        default=1.0,
+    )
+    max_attempts = Integer(
+        display_name='Maximum attempts',
+        help='Defines the number of times a student can try to answer this problem.  If the value '
+        'is not set, infinite attempts are allowed.',
+        scope=Scope.settings,
+    )
+
+    editable_fields = [
+        'display_name',
+        'content',
+        'help_text',
+        'column_widths',
+        'row_heights',
+        'default_tolerance',
+        'max_score',
+        'max_attempts',
+    ]
+
+    # Dictionary mapping cell ids to the student answers.
+    answers = Dict(scope=Scope.user_state)
+    # Dictionary mapping cell ids to Boolean values indicating whether the cell was answered
+    # correctly at the last check.
+    answers_correct = Dict(scope=Scope.user_state, default=None)
+    # The number of points awarded.
+    score = Float(scope=Scope.user_state)
+    # The number of attempts used.
+    attempts = Integer(scope=Scope.user_state, default=0)
+
+    has_score = True
+
+    @property
+    def num_correct_answers(self):
+        """The number of correct answers during the last check."""
+        if self.answers_correct is None:
+            return None
+        return sum(self.answers_correct.itervalues())
+
+    @property
+    def num_total_answers(self):
+        """The total number of answers during the last check."""
+        if self.answers_correct is None:
+            return None
+        return len(self.answers_correct)
+
+    def parse_fields(self):
+        """Parse the user-provided fields into more processing-friendly structured data."""
+        if self.content:
+            self.thead, self.tbody = parse_table(self.content)
+        else:
+            self.thead = self.tbody = None
+            return
+        if self.column_widths:
+            self._column_widths = parse_number_list(self.column_widths)
+        else:
+            self._column_widths = [800 / len(self.thead)] * len(self.thead)
+        if self.row_heights:
+            self._row_heights = parse_number_list(self.row_heights)
+        else:
+            self._row_heights = [36] * (len(self.tbody) + 1)
+
+    def postprocess_table(self):
+        """Augment the parsed table definition with further information.
+
+        The additional information is taken from other content and student state fields.
+        """
+        self.response_cells = {}
+        for row, height in zip(self.tbody, self._row_heights[1:]):
+            row['height'] = height
+            if row['index'] & 1:
```

Comment:
I don't care, so I can go with `% 2` as well.  For me, the intuitive way of thinking about this was "use the last bit", but this isn't the first time that other people's intuition is different. :)